os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -597,22 +605,64 @@ static int reloading_thread(int argc, char *argv[])
 static int update_thread(int argc, char *argv[])
 {
 	int ret;
+	int bin_idx;
+	uint32_t bin_count = binary_manager_get_ucount();
 
 	/* Check whether there is kernel binary for update.
 	 * If kernel update exists, board will be rebooted. Or, it will return negative values. */
 	ret = binary_manager_update_kernel_binary();
-	if (ret != BINMGR_ALREADY_UPDATED) {
-		/* Return errors except for BINMGR_ALREADY_UPDATE which means already the latest kernel binary is running */
+	if (ret != BINMGR_ALREADY_UPDATED && ret != BINMGR_NOT_FOUND) {
+		/* Return errors except for BINMGR_ALREADY_UPDATE which means already the latest kernel binary is running
+		 * and BINMGR_NOT_FOUND means there is no valid binary in inactive partition
+		 */
 		return ret;
 	}
 
+#ifndef CONFIG_USE_BP
+	/* In case of using BP, checking new binaries and updating BP is done when calling binary_manager_set_bootparam.
+	 * When not using BP, check if new common binary or user binaries exist at this time.
+	*/
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	bool need_update = false;
+
+	/* Reload binaries if new binary is scanned */
+	for (bin_idx = 1; bin_idx <= bin_count; bin_idx++) {
+		/* Scan binary files */
+		ret = binary_manager_check_user_update(bin_idx);
+		if (ret == BINMGR_OK) {
+			/* Update index for inactive partition */
+			BIN_USEIDX(bin_idx) ^= 1;
+			need_update = true;
+		} else if (ret == BINMGR_ALREADY_UPDATED || ret == BINMGR_NOT_FOUND) {
```

Comment:
I think ret is one of error value, so else is enough.

Reply:
I separated them to make it easier for users to recognize errors through log message.